classes/ActionScheduler_Store.php

```diff
@@ -133,6 +153,38 @@
 	 */
 	abstract public function get_claim_id( $action_id );
 
+	/**
+	 * Get the last time the action was attempted.
+	 *
+	 * The time should be given in GMT.
+	 *
+	 * @param string $action_id
+	 *
+	 * @return DateTime|null
+	 */
+	abstract public function get_last_attempt( $action_id );
+
+	/**
+	 * Get the last time the action was attempted.
+	 *
+	 * The time should be given in the local time of the site.
+	 *
+	 * @param string $action_id
+	 *
+	 * @return DateTime|null
+	 */
+	abstract public function get_last_attempt_local( $action_id );
+
+	/**
+	 * Set the last attempt for the given action.
+	 *
+	 * @param string $action_id
+	 * @param DateTime $date
+	 *
+	 * @return bool Whether setting the last attempt was successful.
+	 */
+	abstract public function set_last_attempt_date( $action_id, DateTime $date );
```

Comment:
I think this method should probably be renamed `update_last_attempt_date()`.

The issue wioth `set_` is that, at least in the WC CRUD context, it means "set this data on the object in memory" not "save this data to the database", which is what this method actually does.

Initially I was thinking `save_` or `store_` would be a better prefix to match the other methods which write to the database, but I think `update_` is suitable because the action has to exist, so it must have a last attempt date, and that will align it with the wording used in WordPress.

Reply:
Good call. I'm still getting used to thinking about things in context of WC CRUD (even after all these months). I think it's a good idea to change the name to `update_*`.